MediaBrowser.Common/Plugins/PluginManifest.cs

```diff
@@ -0,0 +1,85 @@
+#nullable enable
+using System;
+using MediaBrowser.Model.Plugins;
+
+namespace MediaBrowser.Common.Plugins
+{
+    /// <summary>
+    /// Defines a Plugin manifest file.
+    /// </summary>
+    public class PluginManifest
+    {
+        /// <summary>
+        /// Gets or sets the category of the plugin.
+        /// </summary>
+        public string Category { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the changelog information.
+        /// </summary>
+        public string Changelog { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the description of the plugin.
+        /// </summary>
+        public string Description { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the Global Unique Identifier for the plugin.
+        /// </summary>
+#pragma warning disable CA1720 // Identifier contains type name
+        public Guid Guid { get; set; }
+#pragma warning restore CA1720 // Identifier contains type name
+
+        /// <summary>
+        /// Gets or sets the Name of the plugin.
+        /// </summary>
+        public string Name { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets an overview of the plugin.
+        /// </summary>
+        public string Overview { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the owner of the plugin.
+        /// </summary>
+        public string Owner { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the compatibility version for the plugin.
+        /// </summary>
+        public string TargetAbi { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the upper compatibility version for the plugin.
+        /// </summary>
+        public string MaxAbi { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the timestamp of the plugin.
+        /// </summary>
+        public DateTime Timestamp { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Version number of the plugin.
+        /// </summary>
+        public string Version { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this plugin should be ignored.
+        /// </summary>
+        public PluginStatus Status { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this plugin should automatically update.
+        /// </summary>
+        public bool AutoUpdate { get; set; } = true;
```

Comment:
Can you elaborate on these two? I'm not sure they should be available in the manifest.

Reply:
This is the disk manifest - which is basically the remote manifest plus some fields that control how the plugin operates.

The first is the plugin's status - active, disabled, reboot needed for it to be working, in an error state, or waiting to be deleted on next restart.

The second is a setting that when set to true, will stop the update being automatically updated to the next version by the scheduled task.

Ideally at some point in the future - all the manifest info should go into the db. 

I will correct the description as it's wrong.

As these are system defined - they can't go into the plugin config - as that's controlled by the end user.

Reply:
AutoUpdate is system defined?

Reply:
Disable and AutoUpdate might actually make sense in the basic plugin configuration, so every single plugin inherits those options. System state would lose the "UserDisabled" option but it would make a bit more sense to keep user configurable settings in the same location.

Reply:
System defined as they need to exist outside of the plugin instance, as they are used to determine if the plugin should be instantiated. eg. A plugin's who status is disabled can't be instantiated, in order that its status can be read.

It would be possible to add Autoupdate into the plugin base level - however, this would stop any possibility of auto-updating plugins at startup, as once they are instantiated, a restart is required to cause the unload/reload.